### ISSUE TITLE
Need config ngx-image-croppper with Image drop event .

### ISSUE THREAD

**Issue by Karthickptsm** [MartijnW49/ngx-image-cropper#9]
HI,

How to configure ngx-image-cropper to work with ImageDropEvent

<image-cropper
    [**imageDropEvent**]="imageDropEvent"
    [maintainAspectRatio]="true"
    [aspectRatio]="4 / 3"
    [resizeToWidth]="128"
    format="png"
    (imageCropped)="imageCropped($event)"
></image-cropper>

we are trying to integrate this drag and drop plugin with this crop plugin
https://www.npmjs.com/package/ngx-file-drop

Advanvce Thanks !!!

**Comment by MartijnW49** [comment]
In the readme of ngx-file-drop, I see that they have an Output called "onFileDrop". You should assign the result of that to the "imageChangedEvent" of the image cropper.

**Comment by knexp** [comment]
That doesn't work @MartijnW49

**Comment by MartijnW49** [comment]
It's hard to tell without any code. Could you add some snippets?

**Comment by knexp** [comment]
Thanks for the reply!
For Configuring this image cropper plugin,

<image-cropper
    [imageChangedEvent]="imageChangedEvent"
    [maintainAspectRatio]="true"
    [aspectRatio]="4 / 3"
    [resizeToWidth]="128"
    format="png"
    (imageCropped)="imageCropped($event)"
></image-cropper>

imageChangedEvent: any = '';
croppedImage: any = '';

fileChangeEvent(event: any): void {
    this.imageChangedEvent = event;
}

the above code is  working fine on "Choosing a file from browser event".

We want to configure for drag and drop file event by like this,
https://www.npmjs.com/package/ngx-file-drop

so we have used
<image-cropper
    [imageChangedEvent]="imageChangedEvent"
    [maintainAspectRatio]="true"
    [aspectRatio]="4 / 3"
    [resizeToWidth]="128"
    format="png"
    (imageCropped)="imageCropped($event)"
></image-cropper>

public dropped(event: UploadEvent) {
    this.files = event.files;
    **this.imageChangedEvent = event;**
    for (var file of event.files) {
      file.fileEntry.file(info => {
        console.log(info);
      });
    }
  }

basically we are trying to make this crop plugin to work with image drag and drop.
so is it possible for you to provide  [**imageDropEvent**] for configuring cropper plugin to work with drop event.
So it will look like
<image-cropper
    [imageDropEvent]="imageDropEvent"
    [maintainAspectRatio]="true"
    [aspectRatio]="4 / 3"
    [resizeToWidth]="128"
    format="png"
    (imageCropped)="imageCropped($event)"
></image-cropper>



Thanks again for your Support.!

**Comment by MartijnW49** [comment]
The image-cropper doesn't have a `imageDropEvent` Input. 
But it I think you can get it to work like this:
```
public dropped(event: UploadEvent) {
        this.imageChangedEvent = event;
}
```

The cropper itself should still use the `imageChangedEvent` Input:
```
<image-cropper
    [imageChangedEvent]="imageChangedEvent"
    [maintainAspectRatio]="true"
    [aspectRatio]="4 / 3"
    [resizeToWidth]="128"
    format="png"
    (imageCropped)="imageCropped($event)"
></image-cropper>
```

**Comment by Karthickptsm** [comment]
We try that too, but it didn't work.
Could you please give any other solution or suggestion.

**Comment by MartijnW49** [comment]
I tried to add the ngx-file-drop myself, but I can't get it to work:
```
Error: Uncaught (in promise): Error: StaticInjectorError[NgZone]: 
  StaticInjectorError[NgZone]: 
    NullInjectorError: No provider for NgZone!
````
Which version of Angular are you using?

**Comment by Karthickptsm** [comment]
Our Project Angular Details

Angular CLI: 1.5.0
Node: 8.9.3
OS: linux x64
Angular: 5.1.2
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.5.0
@angular/service-worker: 1.0.0-beta.16
@angular-devkit/build-optimizer: 0.0.36
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.42
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.8.0
@schematics/angular: 0.1.11
@schematics/schematics: 0.0.11
typescript: 2.5.3
webpack-bundle-analyzer: 2.9.1
webpack: 3.8.1

**Comment by MartijnW49** [comment]
I found a way to get it to work, simply set the `dropped` function to this:
```
public dropped(event: UploadEvent) {
        event.files[0].fileEntry.file(
            (ev) => {
                this.imageChangedEvent = {target: {files: [ev]}}
            },
            () => console.log('Failed to load image')
        );
}
```
It isn't the prettiest code, but it works :)
(Note: if you drop multiple files, it will only take the first one)

**Comment by knexp** [comment]
thanks @Mawi137 :+1:

**Comment by Karthickptsm** [comment]
Thanks Mawi137

**Comment by lasantha57** [comment]
```
<div class="cropArea" [ngClass]="{'dotted-boarder': !isFileSelected}" (drop)="dropHandler($event)"
    (dragover)="allowDrop($event)">
    <input #imageUpload type="file" accept="image/*" (change)="onFileChanged($event)" class="hidden">
    <div *ngIf="!isFileSelected" class="select-btn-div">
      <button class="btn btn-primary"
        (click)="imageUpload.click();">Select</button>
    </div>
    <!-- <img-cropper #cropper [image]="imageData" [settings]="cropperSettings"></img-cropper> -->
    <image-cropper #cropper [hidden]="!isFileSelected" [imageChangedEvent]="imageChangedEvent"
      [imageBase64]="croppedImage" (imageLoaded)="imageLoaded()" (imageCropped)="imageCropped($event)"
      [maintainAspectRatio]="true" [resizeToWidth]="350" [roundCropper]="true">
    </image-cropper>
  </div>
```

```
private readImageFile(file: any) {
    const reader = new FileReader();
    reader.onload = (e: any) => {
      this.selectedFileName = file.name;
      this.croppedImage = e.target.result;
    };
    reader.readAsDataURL(file);
  }

  allowDrop(event: any) {
    event.preventDefault();
  }

  //drag and drop image select
  dropHandler(event: any) {
    event.preventDefault();
    if (!this.isFileSelected && event.dataTransfer.items.length > 0 &&
      event.dataTransfer.items[0].kind === 'file' &&
      event.dataTransfer.items[0].type.indexOf('image') > -1) {
      this.isFileSelected = true;
      this.readImageFile(event.dataTransfer.items[0].getAsFile());
    } else {
      return false;
    }
  }
```

Finally, this works fine in my case in both ways (select and d & d). Sorry for the code is not much organized. Simply I converted the file into `Base64` and assigned to `[imageBase64]`.

**Comment by meet0929** [comment]
add some code in the dropped dunction 

```
 public dropped(files: NgxFileDropEntry[]) {
    this.files = files;
    for (const droppedFile of files) {

      // Is it a file?
      if (droppedFile.fileEntry.isFile) {
        const fileEntry = droppedFile.fileEntry as FileSystemFileEntry;
        fileEntry.file(
          (ev) => {
            this.imageChangedEvent = { target: { files: [ev] } }
          }
        );
      }

    }
  }
```
